### PR TITLE
Header shadow and password

### PR DESCRIPTION
**Pull request by ItisEeron**
### Password Validator should account for Password Manager Accounts 
Now the regex used to validate passwords allows more characters. Specific special characters are permitted and should be accepted as valid passwords. Characters such as : <  or > will not be accepted as to prevent injection attacks. 

### Search Form
I wrote a new component in the elements folder. It takes all of its properties from input forms and adds some for itself. I was required to write a filter function since we were never given any idea of how they wanted us to filter through the requests. The panel next to the search was has not been written. I'm leaving this for Sarah since it most likely involves refactoring the Service Request Page.  All new features included in this pull has automated tests written for them and documentation. 

### Holding off on Header Shadow
I spent hours trying to figure this out for web. I figured out the some possible causes: Smart Components, Web Routes, withSceneHeader. However, this was taking way too much time so I decided to put this task on hold. I separated the styles into platform specific files so in the future, (probably the last sprint) maybe we can address this issue.

### CODE REVIEW

**Review by YellowRainBoots — CHANGES_REQUESTED** [review]
Service Requests on web has two navigation bars. Definitely can't stay like this
Screen Shot 2020-05-20 at 2 12 51 PM

**Comment by ItisEeron** [comment]
Was that the only bug. I just ran across this in a different branch btw?

**Comment by YellowRainBoots** [comment]
I haven't been able to replicate this bug on master or develop myself, so I'm wondering if it's isolated to your branch / maybe it'd be resolved if you pulled develop? I didn't test android native too much for yours because it never lowers the screen below the modal / the modal was always in the right place on develop, but I believe it was fine. I can double check though.

**Comment by ItisEeron** [comment]
It's in my branch. I fixed it this morning. I was working on a one of this branch's child branch.
I just recommitted btw. Things should good.

File: homepairs/homepairsUI/src/modals/property-modals/EditTenantModal/EditTenantModal.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ function setInputStyles(){
                 paddingHorizontal: BaseStyles.MarginPadding.mediumConst,
         },
         modalContainer: {
-            flex: 1,
+            flex: Platform.OS === 'web' ? undefined : 1,
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/modals/property-modals/EditTenantModal/EditTenantModal.tsx:41]: you used a conditional flexGrow for the appliance modal's scrollContentContainerStyle + always flex: 1 for its modalContainer, whereas for the tenant modal you made the flex for the modal Container conditional, you sure this is what you wanted?

File: homepairs/homepairsUI/src/utility/SyntaxVerification.tsx [hunk 2]
```diff
@@ -30,13 +30,11 @@ export function isPhoneNumberValid(phone:string){
...
- * Passwords for homepairs only be alphanumerica and they are 
- * between 6 and 25 characters long. This is subject to change in 
- * the future
- * 
+ * Passwords for homepairs only be alphanumeric with some special characters and they are 
+ * between 6 and 50 characters long. This is subject to change in the future
  */
 export function isPasswordValid(input:string) : boolean {
-    const reg = new RegExp("^([A-Za-z0-9]{6,25})$");
+    const reg = new RegExp("^([a-zA-Z0-9!@#$%^&*\-]{8,32})$");
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/utility/SyntaxVerification.tsx:37]: we might need to update the error message if min password is 8 characters now

**Review by YellowRainBoots — APPROVED** [review]
Approved otherwise, just two comments

**Review by YellowRainBoots — APPROVED** [review]
Good comments! I'll pull the search bar stuff into a feature branch.

**Pull request merged**